packages/vite/src/node/baseEnvironment.ts

```diff
@@ -0,0 +1,103 @@
+import colors from 'picocolors'
+import type { Logger } from './logger'
+import type { ResolvedConfig, ResolvedEnvironmentOptions } from './config'
+import type { EnvironmentPlugin } from './plugin'
+
+export class PartialEnvironment {
+  name: string
+  config: ResolvedConfig
+  options: ResolvedEnvironmentOptions
+  logger: Logger
+
+  constructor(
+    name: string,
+    config: ResolvedConfig,
+    options: ResolvedEnvironmentOptions = config.environments[name],
+  ) {
+    this.name = name
+    this.config = config
+    this.options = options
+    const environment = colors.dim(`(${this.name})`)
+    const colorIndex =
+      [...environment].reduce((acc, c) => acc + c.charCodeAt(0), 0) %
+      environmentColors.length
+    const infoColor = environmentColors[colorIndex || 0]
+    this.logger = {
```

Comment:
I think we should keep this one lazy as it was before so it is cheaper to create the `PartialEnvironment` if we don't end up using the logger

Reply:
I can get it back, but I think this is fine, we will only create this PartialEnvironment a few times in testing and at runtime the logger will probably be used everytime

Reply:
Ok, let's merge it as is and we can later on bring it back if needed